crates/tabby-common/src/index.rs

```diff
@@ -101,6 +98,38 @@ impl CodeSearchSchema {
                 .map(|x| Term::from_field_text(self.field_body, x)),
         ))
     }
+
+    pub fn git_url_query(&self, mut git_url: &str) -> Vec<Box<TermQuery>> {
```

Comment:
this won't work: git_url field is indexed as a whole token, thus can only matched against exact url.

I suggest to use a normalization step by matching git_url against all possible git_urls (from db), and used the matched git_url in tantivy query.